jetty-client/src/main/java/org/eclipse/jetty/client/util/FutureResponseListener.java

```diff
@@ -107,6 +107,8 @@ public ContentResponse get(long timeout, TimeUnit unit) throws InterruptedExcept
         boolean expired = !latch.await(timeout, unit);
         if (expired)
             throw new TimeoutException();
+        if (failure instanceof TimeoutException)
+            throw (TimeoutException)failure;
```

Comment:
I don't think this is right. `TimeoutException` thrown from this method means the timeout specified by the caller expired.
We don't want to confuse this semantic with a possible execution exception that also happens to be a `TimeoutException`, but for other reasons. I think this change should be reverted.